docs/source/api.rst

```diff
@@ -134,6 +134,7 @@ Other Algorithm Functions
    retworkx.graph_greedy_color
    retworkx.digraph_union
    retworkx.metric_closure
+   retworkx.steiner_tree
```

Comment:
Is the `Tree` section of the docs a better place for this?

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/392/commits/c5127a2de849c53ed999299f47b6cc5b4d3f96d8